agent/exec/containerd/adapter.go

```diff
@@ -193,6 +214,29 @@ func (c *containerAdapter) isPrepared() bool {
 	return c.container != nil && c.task != nil
 }
 
+func cniConfig(n *api.Network) (*libcni.NetworkConfig, error) {
+	if n.DriverState.Name != "cni" {
```

Comment:
Oops, `cni.ValidateNetworkSpec` is looking at `DriverConfig` not `DriverState`, so `cniNetworkAllocator.Allocate` isn't redundant, I should still add the check here though.